lib/beaker/options/options_hash.rb

```diff
@@ -60,14 +62,18 @@ def is_pe?
       #     a.get_type == :pe
       #
       # @return [Symbol] the type given in the options
-      def get_type
+      def get_type name
         case self[:type]
         when /pe/
           :pe
         when /foss/
           :foss
         when /aio/
-          :aio
+          if self['HOSTS'][name]['roles'].include? 'agent'
```

Comment:
This doesn't belong in OptionsHash - which is just an extension of hash and should not have included information as to the structure of what it is storing.  In some cases it will not have a 'HOSTS' section and then this will die.